Realm/Realm/DataBinding/RealmObjectTypeDelegator.cs

```diff
@@ -45,7 +45,18 @@ protected override PropertyInfo GetPropertyImpl(string name, BindingFlags bindin
                 return null;
             }
 
-            if (_schema.Any(p => p.ManagedName == name))
+            if (_schema != null)
+            {
+                if (_schema.Any(p => p.ManagedName == name))
+                {
+                    return _propertyCache.GetOrAdd(name, n => new WovenPropertyInfo(result));
+                }
+
+                return result;
+            }
+
+            // Schema is null only for woven unmanaged objects
+            if (result?.GetCustomAttribute<WovenPropertyAttribute>() != null)
             {
                 return _propertyCache.GetOrAdd(name, n => new WovenPropertyInfo(result));
             }
```

Comment:
All of that is very repetitive :D

Can we rework to something like:

```suggestion
        if (_schema?.Any(p => p.ManagedName == name) == true || result.GetCustomAttribute<WovenPropertyAttribute>() != null)
        {
            return _propertyCache.GetOrAdd(name, n => new WovenPropertyInfo(result));
        }
```

Reply:
It is repetitive, but it's also slightly more efficient. If `schema != null`, then we don't need to look for the `WovenPropertyAttribute` anyways, as the schema will always have all the necessary info.
What you wrote was my first version, but I thought it would make sense to be a little more efficient with this if we can.

Reply:
It is repetitive, but it's also slightly more efficient. If schema != null, then we don't need to look for the WovenPropertyAttribute anyway, as the schema will always have all the necessary info.
What you wrote was my first version, but I thought it would make sense to be a little more efficient with this if we can.

Reply:
I am curious, would the other "or" statements be evaluated if the first one? I would have assumed if the first condition i.e. `_schema?.Any(p => p.ManagedName == name) == true` is true then it would not bother to evaluate others, thus functionally being similar to the more explicit, longer version.

Reply:
You are correct, if the first condition is true then it would not check the second one. Here the situation is a little different though.
If the schema is not null and the property is found then the second condition is not evaluated and it's all good.
If the schema is not null and the property is not found, then we don't need to check the second condition anyway because we have all the properties in the schema and it will be useless. 
Only if the schema is null we should check the second condition.

Reply:
In C# `&&` and `||` are short-circuiting, so what Nikola suggested is as efficient as the longer version.
If you specifically want to avoid the short-circuiting you need to use `&` and `|`.

Reply:
That was it! I didn't think about this 😁